packages/playground/windows/playground/HostingPane.xaml.cpp

```diff
@@ -318,6 +319,22 @@ void HostingPane::OnSelectionChanged_JavaScriptFilename(
   UpdateUI();
 }
 
+void HostingPane::OnTextChanging_ValidPort(TextBox ^ sender, TextBoxTextChangingEventArgs ^ args) {
+  std::wstring buffer = L"0";
+  for (auto wchar = sender->Text->Begin(); wchar != sender->Text->End(); ++wchar) {
+    if (iswdigit(*wchar)) {
+      buffer.push_back(*wchar);
+    }
+  }
+
+  auto port = std::stoi(buffer);
+  if (port > UINT16_MAX) {
+    port = UINT16_MAX;
+  }
+
+  sender->Text = ref new String(std::to_wstring(port).c_str());
+}
```

Comment:
It looks like this can be simplified to set args.Cancel = true if the character being processed is not a digit. See [https://blog.mzikmund.com/2019/01/number-only-textbox-in-uwp/](https://blog.mzikmund.com/2019/01/number-only-textbox-in-uwp/)

Reply:
This _should_ be the default when InputScope=Number but apparently it isn't? #sad.

---
In reply to: [397537756](https://github.com/microsoft/react-native-windows/pull/4373#discussion_r397537756) [](ancestors = 397537756)

Reply:
That would simplify so much, thanks for sharing!
Unfortunately, it seems that this API is not supported on Windows 10 versions before v10.0.16299.0 (I think the current minimum supported version is v10.0.10240.0). What do you suggest?

Reply:
bummer that we don't get beforetextchanging earlier. I don't think we need to support 10240 which is TH1. I think the earliest, at most, should be RS2 (15063), but maybe we can push it to RS3 which is 16299. One idea is to use the event if running on 16299+ and just not worry about it if running on earlier OS (treat it as invalid input and use the default port)

Reply:
You are right, the minimum supported version was RS2 (15063). The minimum supported version was updated to RS3 (16299) in #4449 so I have updated the TextBox to utilize the event.
Otherwise I would've went with the other approach which also made a lot of sense.
Thanks!